### PR TITLE
Added batching support & made inference 2.66x faster

### PR DESCRIPTION
**Pull request by AlexandreBrown**
I added support for computing the image embeddings for a batch of images that are already preprocessed.  
This allows much faster processing since now we can call `set_image_batch` once for all images instead of once for every image in a foor loop.    
The choice of not doing preprocessing inside `set_image_batch` aligns with its goal of being fast since `set_image` expects a cpu numpy array in uint8 and requires to do various conversions that might not be necessary if someone already has a tensor in float32 already normalized in the range (0,1).  
I also added `apply_boxes_torch` which allows preprocessing of the bounding box in pure pytorch (no cuda torch.Tensor to cpu numpy conversion just to finally go back to cuda torch.Tensor).    
  
Test using a batch of 128 images (128, 3, 512, 512)  using pretrained `efficientvit_sam_l0`
**Before**   
```python
numpy_images = rearrange(images, "b c h w -> b h w c").cpu().numpy()

for i, (numpy_image, image_bounding_boxes) in enumerate(
    zip(numpy_images, images_bounding_boxes)
):
    self.segments_predictor.set_image(numpy_image) # Computes image embeddings 1 image at atime

    preprocessed_image_boxes = self.preprocess_boxes(image_bounding_boxes) # Using apply_boxes

    binary_masks, _, _ = self.segments_predictor.predict_torch(
        point_coords=None,
        point_labels=None,
        boxes=preprocessed_image_boxes,
        multimask_output=False,
    )
```
Time on a Quadro RTX 8000 : ~8s  

**After**  
```python
self.segments_predictor.set_image_batch(sam_images) # sam_images are cuda torch.Tensor (B,C,H,W) float32 in range (0,1) already

for image_index, image_bounding_boxes in enumerate(images_bounding_boxes):
    preprocessed_image_boxes = self.preprocess_boxes_for_sam(
        image_bounding_boxes
    ) # Uses apply_boxes_torch

    image_binary_masks, _, _ = self.segments_predictor.predict_torch(
        image_index=image_index,
        point_coords=None,
        point_labels=None,
        boxes=preprocessed_image_boxes,
        multimask_output=False,
    )
```  
Time on a Quadro RTX 8000: ~3s

### CODE REVIEW

**Comment by zhuoyang20** [comment]
Thanks for your contribution! This is highly beneficial for batch processing. I agree to merge it into the main branch.
